js/packages/web/src/contexts/tokenList.tsx

```diff
@@ -36,27 +36,29 @@ export function SPLTokenListProvider({ children = null as any }) {
       ...process.env.NEXT_SPL_TOKEN_MINTS.split(",")
     ]: [WRAPPED_SOL_MINT]
 
+  const { tokens } = useConnectionConfig();
+
   useEffect(() => {
-    new TokenListProvider().resolve().then(setTokenList);
-  }, [setTokenList]);
+    setTokenList(new TokenListContainer(Array.from(tokens.values())))
```

Comment:
This is the main change to fetch the list of tokens from the already-filtered by chain id `tokens` (coming from the connection) instead of the full list of tokens.